src/components/flyout/deviceReconfigureFlyout.js

```diff
@@ -32,14 +42,46 @@ class DeviceReconfigureFlyout extends React.Component {
     this.commonConfigValueChanged = this.commonConfigValueChanged.bind(this);
     this.onChangeInput = this.onChangeInput.bind(this);
     this.applyDeviceConfigureJobsData =this.applyDeviceConfigureJobsData.bind(this);
+    this.checkJobStatus = this.checkJobStatus.bind(this);
   }
 
-  componentWillMount() {
-    this.calcCommonConfiguration(this.props.devices);
+  componentDidMount() {
+    const { devices, propertyUpdateJobs } = this.props;
+    this.checkJobStatus(devices, propertyUpdateJobs);
+    this.calcCommonConfiguration(devices);
   }
 
   componentWillReceiveProps(nextProps) {
-    this.calcCommonConfiguration(nextProps.devices);
+    const { devices, propertyUpdateJobs } = nextProps;
+    if (!_.isEqual(propertyUpdateJobs, this.props.propertyUpdateJobs)) {
+      this.checkJobStatus(devices, propertyUpdateJobs);
+    }
+    this.calcCommonConfiguration(devices);
+  }
+
+  checkJobStatus (devices, propertyUpdateJobs) {
+    const jobs = getRelatedJobs(devices, propertyUpdateJobs);
+    const jobStream = Rx.Observable.from(jobs);
```

Comment:
No real need to save this as its own variable. I would just use the observable directly.